Disallow unquoted literals in `LIKE` clause in `DESCRIBE` statement

Yury-Fridlyand opened the pull request:
### Description
Update SQL parser to disallow unquoted strings as identifiers wildcards clause.

To test this fix for #650 disable legacy engine first:
```diff
diff --git a/legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java b/legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.>
index ab146404f..29c501117 100644
--- a/legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java
+++ b/legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java
@@ -161,10 +161,11 @@ public class RestSqlAction extends BaseRestHandler {
                             QueryContext.getRequestId());
                         result.accept(channel);
                     } else {
-                        LOG.debug("[{}] Request {} is not supported and falling back to old SQL engine",
+                        LOG.info("[{}] Request {} is not supported and falling back to nowhere",
                             QueryContext.getRequestId(), newSqlRequest);
-                        QueryAction queryAction = explainRequest(client, sqlRequest, format);
-                        executeSqlRequest(request, queryAction, client, channel);
+                        throw new SQLFeatureDisabledException("buuuu");
+                        //QueryAction queryAction = explainRequest(client, sqlRequest, format);
+                        //executeSqlRequest(request, queryAction, client, channel);
                     }
                 } catch (Exception e) {
                     logAndPublishMetrics(e);
(
```

**Before**
_With patch, without fix_
```sql
opensearchsql> DESCRIBE TABLES LIKE flights COLUMNS LIKE time%;
{'reason': 'Invalid SQL query', 'details': 'buuuu', 'type': 'SQLFeatureDisabledException'}
opensearchsql> DESCRIBE TABLES LIKE flights COLUMNS LIKE %time%;
{'reason': 'Invalid SQL query', 'details': 'buuuu', 'type': 'SQLFeatureDisabledException'}
opensearchsql> DESCRIBE TABLES LIKE calcs;
Output longer than terminal width
Do you want to display data vertically for better visual effect? [y/N]
```

**After**
```sql
opensearchsql> DESCRIBE TABLES LIKE flights COLUMNS LIKE time%;
{'reason': 'Invalid SQL query', 'details': 'buuuu', 'type': 'SQLFeatureDisabledException'}
opensearchsql> DESCRIBE TABLES LIKE flights COLUMNS LIKE %time%;
{'reason': 'Invalid SQL query', 'details': 'buuuu', 'type': 'SQLFeatureDisabledException'}
opensearchsql> DESCRIBE TABLES LIKE calcs;
{'reason': 'Invalid SQL query', 'details': 'buuuu', 'type': 'SQLFeatureDisabledException'}
```

Another queries for test:
```sql
DESCRIBE TABLES LIKE acc%;
SHOW TABLES LIKE %;
```
and
```sql
DESCRIBE TABLES LIKE 'acc%';
SHOW TABLES LIKE "%";
```

### Notes
As discussed, we should prohibit usage of raw identifies in SQL queries where wildcards are allowed. A user should specify string literal instead (a quoted string with `'` or `"`). Backtick quote `` ` `` is not accepted.
This behavior was copied from legacy engine and it is an erroneous approach.
The root cause of #650 is a `keywordsCanBeId` which was in the user query which cased V2 parser to fail and fall back to V1.

Breaking change for 3.x release. Don't backport.

See team review and discussion in https://github.com/Bit-Quill/opensearch-project-sql/pull/154.

### Issues Resolved
Fixes #650
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

penghuo commented:
The [syntax section](https://github.com/Bit-Quill/opensearch-project-sql/blob/integ-unquoted-percent-in-like-clause/docs/user/dql/metadata.rst#syntax) should be changed.

Yury-Fridlyand commented:
> The [syntax section](https://github.com/Bit-Quill/opensearch-project-sql/blob/integ-unquoted-percent-in-like-clause/docs/user/dql/metadata.rst#syntax) should be changed.

Rebased and updated in ac4875f4f.

dai-chen reviewed (approved):
Thanks for the fix!
